### PR TITLE
EC-54 and EC-55

### PR DESCRIPTION
**Pull request by tboinske-skyhook**
EC-54: Pull a fix in from develop branch
EC-55: Implement virtual MAC filtering and AP down-selection

### CODE REVIEW

File: src/client/ap_filter.c [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        if (j != n)
+            memcpy(&aps[n], &aps[j], sizeof(struct ap_t));
+    }
+
+    memcpy(&aps[MAX_FILTERED_APS-1], &aps[num_aps-1], sizeof(struct ap_t));
+
+    return MAX_FILTERED_APS;
+}
+
+size_t filter_and_reduce_aps(struct ap_t aps[], size_t num_aps)
```
Comment (tboinske-skyhook) [inline src/client/ap_filter.c]: @syarramsetty-skyhook - Please review this function, and this file. The other changes in this PR are just simple refactorings and cleanups, and don't require any special attention on your part.

File: src/client/ap_filter.c [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+    mac_to_hex_str(sa, macA);
+    mac_to_hex_str(sb, macB);
+
+    size_t num_diff = 0; // Num hex digits which differ
+    size_t i;
+
+    for (i = 0; i < MAC_HEX_STR_SIZE; i++)
+        if (sa[i] != sb[i] && ++num_diff > 1)
+            return 0;
```
Comment (syarramsetty-skyhook) [inline src/client/ap_filter.c]: Are they similar If only one `hex` digit is different in `OUI`?

File: src/client/ap_filter.c [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+#if defined UNIT_TEST
+// gcc -DUNIT_TEST -I../../common/inc/ ../../bin/sky_util.o ap_filter.c
+
+void sky_set_ap(struct ap_t * ap,
+        int8_t rssi, uint8_t * mac,
+        bool is_connected, enum SKY_BAND band) {
+    ap->rssi = rssi;
+    strncpy((char *)ap->MAC, (const char *)mac, 6);
```
Comment (syarramsetty-skyhook) [inline src/client/ap_filter.c]: `6 -> MAC_SIZE`

Comment (tboinske-skyhook) [reply src/client/ap_filter.c]: Thanks. Changed.

**Comment by tboinske-skyhook** [comment]
Closing this PR in favor of #7 (which represents the same changes, but all contained within a single commit).

**Pull request closed without merging**